hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/RMWebServices.java

```diff
@@ -734,7 +739,80 @@ public Response updateAppState(AppState targetState,
 
     return Response.status(Status.OK).entity(ret).build();
   }
-  
+
+  // Copy-paste job from updateAppQueue() above . Almost exactly the same pattern as YARN-1702.
```

Comment:
Indeed, it looks very similar. Can some of this security dance be factored into a common method?